### PR TITLE
Adding a eslintrc configuration that can be run from npm. Cleaning up…

### PR DESCRIPTION
**Pull request by rowasc**
**TLDR: WIP - is this something that you would be interested in including to the repo? (eslint + npm lint task)** 

Hi, 
I noticed there's a PR by @kkota1 that solves a lot of code style issues, and figured it'd be a good idea to add a linter to the repo so we can clean up with some consistent rules (also, because this way we can also avoid at least _some_ basic errors while we don't have a test suite). My idea was to start by fixing the automatic things that we can detect, and merge with that other PR for the specific fixes introduced there (like require statements that change, and some arrow functions)

Anyway, here's a preview of the work (still a WIP!). Do you think this is something that could be merged once it's finished? asking so I know if it's worth it to invest more time into it (I should have opened an issue beforehand.. now that I think about it.. )

After this, adding a test suite would be awesome, but a linter is a _super_ cheap starting point for consistent quality. I'm not so interested in the specific settings as I'm in having a code style, so I used a standard setup from node and went from there (with the change of having 4 space indents since that's what I noticed you were using here)

### CODE REVIEW

**Comment by zeyarmin007** [comment]
#47

**Comment by jeromewu** [comment]
This is great but we are working on newer version of the eslint, please check develop branch for more information.
